### PR TITLE
add psbt scan to sign flow happy path

### PR DESCRIPTION
**Pull request by newtonick**

### CODE REVIEW

**Comment by kdmukai** [comment]
Awesome! I suspect that we'll want to eventually make your test psbt a reusable resource that can be applied to other tests.

Also note that I just added some breaking changes:
https://github.com/SeedSigner/seedsigner/pull/339#issuecomment-1509166278

**Comment by jdlcdl** [comment]
@newtonick: just a heads-up to avoid spinning wheels if you end up seeing the following warnings as you get through your flow tests.

```
File "seedsigner-dev/src/seedsigner/gui/components.py", line 316, in __post_init__
   if not self.auto_line_break or full_text_width < self.supersampled_width - (2 * self.edge_padding self.supersampling_factor):
TypeError: '<' not supported between instances of 'int' and 'MagicMock'
warnings.warn(pytest.PytestUnhandledThreadExceptionWarning(msg))
```

Keith figured this out this morning after I struggled/floundered to no avail:

The LoadingScreenThread needs to be mocked for flow tests, and it's already patched in tests/base.py, however if the views are importing this class at the module level, instead of within SomeClass.run(), then the real LoadingScreenThread is apparently getting imported before being patched with MagicMock.

I suspect that this warning will appear as flow tests are written for PSBTChangeDetailsView and PSBTOverviewView.

**Comment by newtonick** [comment]
I believe I have this working and it is ready for review

File: src/seedsigner/views/psbt_views.py [hunk 1]
```diff
@@ -128,22 +131,27 @@ def run(self):
...
-            num_inputs=psbt_parser.num_inputs,
-            num_self_transfer_outputs=num_self_transfer_outputs,
-            num_change_outputs=num_change_outputs,
-            destination_addresses=psbt_parser.destination_addresses,
-        )
+        spend_amount = psbt_parser.spend_amount
+        change_amount = psbt_parser.change_amount
+        fee_amount = psbt_parser.fee_amount
+        num_inputs = psbt_parser.num_inputs
+        destination_addresses = psbt_parser.destination_addresses
```
Comment (kdmukai) [inline src/seedsigner/views/psbt_views.py:138]: Minor note: doesn't seem to be any reason to pull these out into local vars when the values can just be directly passed in below in lines 147-153.

**Comment by kdmukai** [comment]
Test is passing, changes look great. If anything, I'd say that the test is *over-documented*! But that's not necessarily a bad thing.

Just the minor note above, but otherwise love it!

**Pull request merged**